### PR TITLE
Make the importer a typed actor

### PR DESCRIPTION
**Pull request by dominiklohmann**
###  :notebook_with_decorative_cover: Description

This converts the IMPORTER component to the `caf::typed_actor` API.

###  :memo: Checklist

- [x] All user-facing changes have changelog entries.
- [x] The changes are reflected on [docs.tenzir.com/vast](https://docs.tenzir.com/vast), if necessary.
- [x] The PR description contains instructions for the reviewer, if necessary.

### :dart: Review Instructions

File-by-file.

### CODE REVIEW

File: libvast/vast/system/actors.hpp [hunk 1]
```diff
@@ -281,6 +281,10 @@ using exporter_actor = typed_actor_fwd<
   // Register the SINK actor.
   caf::reacts_to<atom::sink, caf::actor>,
   // Register a list of IMPORTER actors.
+  // FIXME: This cannot take an importer_actor, because that'd introduce a
+  // cyclic dependency between the importer_actor and exporter_actor
+  // declarations. We should improve the logic of wiring up the importers with
+  // continuous exporters, e.g., by having the node deal with the issue.
```
Comment (tobim) [inline libvast/vast/system/actors.hpp:287]: I implemented a solution to this. Am I allowed to push on this branch?

Comment (dominiklohmann) [reply libvast/vast/system/actors.hpp:287]: Yes, I just turned off auto-merge.

Comment (tobim) [reply libvast/vast/system/actors.hpp]: I just did, let me know what you think.

Comment (dominiklohmann) [reply libvast/vast/system/actors.hpp]: Yes, that works.

I'll do another pass to fixup the naming so it's more consistent.

Comment (dominiklohmann) [reply libvast/vast/system/actors.hpp]: Also just noticed that your change broke CI, so you must've missed some non-typed actor messaging.

**Review by tobim — APPROVED** [review]
Good start, now let's get the other stuff in too.

**Pull request merged**